src/libprojectM/PCM.cpp

```diff
@@ -338,7 +338,6 @@ void PCM::getPCM(float *PCMdata, int samples, int channel, int freq, float smoot
    if (freq)
      {
        // NOTE some presets set bSpectrum=1 and samples!=2^n, not sure what rdft() does with that
-       assert(samples <= 1024);
```

Comment:
https://github.com/projectM-visualizer/projectm/pull/312/files#r571438718
This is causing a crash for me, I'm not sure what the solution is here, but the next line seems to handle it sort of gracefully? @mbellew

Reply:
After removing this line, I can trigger a crash in the fft code :(
```
Process 93461 stopped
* thread #1, name = 'projectMSDL', stop reason = signal SIGSEGV: invalid address (fault address: 0x7fff9ac26d98)
    frame #0: 0x00005555555838e4 projectMSDL`cftmdl1(n=540, a=0x00007fffffffa0e0, w=0x0000555555a79cc0) at fftsg.cpp:2505:19
   2502	        x0r = x1r + x3i;
   2503	        x0i = x1i - x3r;
   2504	        a[j3] = wk3i * x0r + wk3r * x0i;
-> 2505	        a[j3 + 1] = wk3i * x0i - wk3r * x0r;
   2506	    }
   2507	    j0 = mh;
   2508	    j1 = j0 + m;
```

Reply:
Possible fix: 8e46235f